students/Hovhannes_Muradyan/Lesson1/hw1.txt

```diff
@@ -0,0 +1,99 @@
+Задача 1.
+С помощью битовых операций проверить число на четность 
+примеры 7 -> false, 4578 -> true
+bool is_even(int x)
+{
+	return !(x&1);
+}
+
+
+
+
+
+
+
+Задача 2.
+Посчитать количество нулей в начале двоичного представления:
+пример x = 784252715 (00101110101111101011111100101011) -> функция должна возвращать 2
+int nulls_count(int x)
+{
+  int counter = 0;
+  for(int i = 30; i >=0 ; i--){
+    if(x & (1<<i)){
+      return counter;
+    }
+    counter++;
+  }
+}
+
+
+
+
+
+
+Задача 3.
+Даны числа x и i, требуется обнулить все биты числа x от самого старшeго до i.
+Пример x = 784252715 (00101110101111101011111100101011), i = 13 -> функция должна возвращать 442155 (00000000000000000001111100101011)
+# В примере ошибка , вместо 442155 должен быть 7979
+int clear_bits_through_i(int x, int i)
+{
+  int a = INT32_MAX;
```

Comment:
here too, the solution is right if int is 32 bit, but it'd be better to replace `INT32_MAX` but `INT_MAX` and 31 by sizeof(int)*8-1